back/app/models/initiative.rb

```diff
@@ -96,9 +96,10 @@ class Initiative < ApplicationRecord
   end)
 
   scope :with_status_code, (proc do |code|
-    joins('LEFT OUTER JOIN initiative_initiative_statuses ON initiatives.id = initiative_initiative_statuses.initiative_id')
-      .joins('LEFT OUTER JOIN initiative_statuses ON initiative_statuses.id = initiative_initiative_statuses.initiative_status_id')
-      .where(initiative_statuses: { code: code })
+    left_outer_joins(:initiative_initiative_status)
+      .where(initiative_initiative_statuses: {
+        initiative_status_id: InitiativeStatus.where(code: code).select(:id)
```

Comment:
```suggestion
        initiative_status_id: InitiativeStatus.where(code: code)
```
If you use a scope returned by `where` instead of an array of ids returned by `select`, it'll be put in the same query as `join` instead of querying the DB 2 times.